### PR TITLE
Erl call take if no erl config

### PR DESCRIPTION
**Pull request by pubalokta**
### Description
This PR includes 3 minor fixes:
- If calling `takeElevated` for a bucket with no elevated configuration, it allows the call to takeElevated.lua but indicating ERL is disabled so it doesn't attempt to activate it. It 
- Adding takeElevated to client.js 
- takeElevated now returns the quota left in the bucket after activating ERL instead of the quota previous to activating it, to make it easier for the clients to understand it.

### Testing

> Describe how this can be tested by reviewers. Be specific about anything not tested and reasons why. If this library has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.
>
> Also include details of the environment this PR was developed in (language/platform/browser version).

- [ x ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ x ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ x ] The correct base branch is being used, if not the default branch

### CODE REVIEW

File: lib/take_elevated.lua [hunk 1]
```diff
@@ -8,6 +8,7 @@ local erl_bucket_size = tonumber(ARGV[7])
 local erl_activation_period_seconds = tonumber(ARGV[8])
 local erl_quota_amount = tonumber(ARGV[9])
 local erl_quota_expiration_epoch = tonumber(ARGV[10])
+local is_erl_enabled = ARGV[11] == "true" and true or false
```
Comment (pubalokta) [inline lib/take_elevated.lua:11]: just in case, this is the ternary operator in lua, similar to
`condition ? true : false`

File: lib/db.js [hunk 2]
```diff
@@ -289,17 +289,24 @@ class LimitDBRedis extends EventEmitter {
...
     }
     const erlParams = utils.getERLKeysQuotaAmountAndExpiration(params.elevated_limits);
 
     this._doTake(params, callback, (key, bucketKeyConfig, count) => {
-      const valError = validateConfigIsForElevatedBucket(key, bucketKeyConfig)
-      if (valError) {
-        return callback(valError)
+      let isERLEnabledForBucket = true;
+      const valERLConfigError = validateConfigIsForElevatedBucket(key, bucketKeyConfig)
+      if (valERLConfigError) {
```
Comment (LeweyM) [inline lib/db.js:301]: is this semantically different than going down the `take()` function at this point?

Comment (LeweyM) [reply lib/db.js:301]: sorry, i know there was a slack thread on this but I found it hard to parse 😅

Comment (LeweyM) [reply lib/db.js:301]: I suppose the return values are different?

Comment (pubalokta) [reply lib/db.js:301]: yes. If we make it go the `take()`, we will only be using `takeElevated.lua` for buckets with elevated limits configuration and we'll always depend on both scripts.
This way we'll be making sure `takeElevated.lua` works for both scenarios and in the future we'll be able to join them in a single method if we wish to do so.

File: lib/db.js [hunk 3]
```diff
@@ -289,17 +289,24 @@ class LimitDBRedis extends EventEmitter {
...
+        isERLEnabledForBucket = false
```
Comment (LeweyM) [inline lib/db.js:302]:
```suggestion
      const valERLConfigError = validateConfigIsForElevatedBucket(key, bucketKeyConfig)
      const isERLEnabledForBucket = !valERLConfigError;
      if (valERLConfigError) {
      
 ```
 
 to reduce code with mutable variables

**Review by LeweyM — APPROVED** [review]
lgtm with minor suggestion 👍

File: lib/db.js (hunk 3 above)
Comment (pubalokta) [reply lib/db.js:302]: I originally wrote it like that, but I thought it would be clearer for the reader to understand that we're disabling ERL after not finding ERLConfig.
Do you find it clear what the intention is this way `const isERLEnabledForBucket = !valERLConfigError `?
If you think it is, I'll apply the change right away!

Comment (LeweyM) [reply lib/db.js:302]: I think it's better. I see what you mean; I think both ways are sufficiently clear, but not having to follow around mutable variables is much easier to follow imo
